test/functional/runner/index.js

```
@@ -0,0 +1,591 @@
+    'unpin after transient error within a transaction and commit',
+    'count'
+  ];
+
+  if (spec.skipReason || SKIP_TESTS.indexOf(spec.description) !== -1) {
+    return it.skip;
+  }
+
+  return it;
+}
+
+function generateTopologyTests(testSuites, testContext) {
+  testSuites.forEach(testSuite => {
+    const environmentRequirementList = parseRunOn(testSuite.runOn);
+
+    environmentRequirementList.forEach(requires => {
+      const suiteName = `${testSuite.name} - ${requires.topology.join()}`;
+      describe(suiteName, {
+        metadata: { requires },
+        test: function() {
+          beforeEach(() => prepareDatabaseForSuite(testSuite, testContext));
+          afterEach(() => testContext.cleanupAfterSuite());
+
+          testSuite.tests.forEach(spec => {
+            const maybeSkipIt = shouldSkipTest(spec);
+            maybeSkipIt(spec.description, function() {
+              let testPromise = Promise.resolve();
+
+              if (spec.failPoint) {
+                testPromise = testPromise.then(() => testContext.enableFailPoint(spec.failPoint));
+              }
+
+              // run the actual test
+              testPromise = testPromise.then(() =>
+                runTestSuiteTest(this.configuration, spec, testContext)
+              );
+
+              if (spec.failPoint) {
+                testPromise = testPromise.then(() => testContext.disableFailPoint(spec.failPoint));
+              }
+
+              return testPromise.then(() => validateOutcome(spec, testContext));
+            });
+          });
+        }
+      });
+    });
+  });
+}
+
+// Test runner helpers
+function prepareDatabaseForSuite(suite, context) {
+  context.dbName = suite.database_name;
+  context.collectionName = suite.collection_name;
+
+  const db = context.sharedClient.db(context.dbName);
+  const coll = db.collection(context.collectionName);
+
+  return db
+    .admin()
+    .command({ killAllSessions: [] })
+    .catch(() => {}) // ignore any error from this
```

Why do we want to ignore errors here?

---

its part of the test runner [specification](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst)

---

I think the spec says to only ignore error code 11601

---

changed!